docs/multicall-notes.md

```diff
@@ -0,0 +1,225 @@
+	"result": [
+		{
+			"number": "0x4",
+			"hash": "0x859c932c5cf0dabf8d12eb2518e063966ac1a25e2fc49f1f02574a37f358d0b5",
+			"timestamp": "0x1f",
+			"gasLimit": "0x4c4b40",
+			"gasUsed": "0x5208",
+			"feeRecipient": "0x0000000000000000000000000000000000000000",
+			"baseFeePerGas": "0x2310a91d",
+			"prevRandao": "0x0000000000000000000000000000000000000000000000000000000000000000",
+			"calls": [
+				{
+					"returnData": "0x",
+					"logs": [
+						{
+							"address": "0x0000000000000000000000000000000000000000",
+							"topics": [
+								"0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef",
+								"0x000000000000000000000000c000000000000000000000000000000000000000",
+								"0x000000000000000000000000c100000000000000000000000000000000000000"
+							],
+							"data": "0x00000000000000000000000000000000000000000000000000000000000003e8",
+							"blockNumber": "0x4",
+							"transactionHash": "0xa4d41019e71335f8567e17746b708ddda8b975a9a61f909bd3df55f4866cc913",
+							"transactionIndex": "0x0",
+							"blockHash": "0x859c932c5cf0dabf8d12eb2518e063966ac1a25e2fc49f1f02574a37f358d0b5",
+							"logIndex": "0x0",
+							"removed": false
+						}
+					],
+					"gasUsed": "0x5208",
+					"status": "0x1"
+				}
+			]
+		}
+	]
+}
+```
+
+Here the interesting part is:
+```json
+"address": "0x0000000000000000000000000000000000000000",
+"topics": [
+	"0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef",
+	"0x000000000000000000000000c000000000000000000000000000000000000000",
+	"0x000000000000000000000000c100000000000000000000000000000000000000"
+],
+"data": "0x00000000000000000000000000000000000000000000000000000000000003e8",
+```
+As can be seen, the sending address is the zero address, `0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef` corresponds to signature `Transfer(address,address,uint256)`, `"0x000000000000000000000000c000000000000000000000000000000000000000"` corresponds the sending address and `0x000000000000000000000000c100000000000000000000000000000000000000` is the receiving address. The amount of ETH moved is stored in the `data` field.
+
+The ETH logs will contain following types of ETH transfers:
+ - Transfering ETH from EOA
+ - Transfering ETH via contract
+ - Selfdestructing contract sending ETH
+
+But not following ones:
+ - Gas fees
+ - Multicalls eth balance override  
+
+ETH logs are not part of the calculation for logs bloom filter. Also, similar to normal logs, if the transaction sends ETH but the execution reverts, no log gets issued.
+
+## Validation
+The multicall has a feature to enable or disable validation with setting `Validation`, by default, the validation is off, and the multicall mimics `eth_call` with reduced number of checks. Validation enabled mode is intended to give as close as possible simulation of real EVM block creation, except there's no checks for transaction signatures and we also allow one to send a direct transaction from a contract.
```

Comment:
I am not entirely sure what this would remove/add. If I turn this on, in what situation would this be useful, and what would change if I would have kept it off? :thinking:

Reply:
For instance, if validation is set to off, are tx gas limits not checked? (So I can include a tx with gas limit higher than the block gas limit?).

Reply:
Never mind, it is in `execute.yaml`

```
    validation:
      title: Validation
      description: |-
        When true, the multicall does all validations that a normal EVM would do, except contract sender and signature checks. When false, multicall behaves like eth_call.
        ```

Reply:
This is bit ill defined (as discussed in todays call). Because Nethermind and Geth behave bit differently with `eth_call` and we have been trying to figure out the differences so we could document them. The intention that is that when validation is enabled, the clients would behave exactly the same, while not fixing the issue that with non-validation, the validation rules might differ a bit.

Reply:
Then I think this spec should explicitly state what should be *not* be validated if the validation is off. For instance, currently it only seems that you can override the sender (so you can create unsigned txs). But, it also seems that it is possible to set base fee to zero (so do not validate this as well). It should be clearly defined what should be validated and what not. (And this is non-trivial since there is a huge ruleset of what should be validated in transactions/blocks)

Reply:
Agreed.  We should start from "validate exactly like mainnet" and then include a list of things that are skipped.  Transaction signatures aren't validated and can be entirely missing, and the no-contract-caller check is skipped.  For block validation, on the call it sounded like we do want to validate that the base fee and gas limit are within bounds of the parent block though.

Reply:
I think we should then put this on the agenda for the next call. Because if we want to validate `base fee` to be within bounds of the parent block, but we also allow for tx base fee to be 0 (which cannot happen on mainnet), then (for me) this is unlogical.

We might want to first create a list of all items which should be validated for txs + blocks. (And this validation list will increase once we hit Cancun :sweat_smile: ). We can then decide what checks we should validate and what not (and possibly also create multiple validation modes / flags if necessary?)

Reply:
For validation mode, I'm fine with base fee not being allowed to go below 7.

Reply:
I don't think it makes sense that much to validate base fee in validation mode if the user specifically sets the base fee to something. Otherwise we could remove all overrides for validation mode, as none of them can happen on mainnet.

Reply:
> I'd personally like to hear from some people to say they need this first.

Our team would like it because it allows us to verify that the transaction is properly constructed and will execute against mainnet without error.  In some cases we ship a set of transactions as a "bundle" to searchers, and they need to all execute properly in sequence (including things like balance checks), and in other cases we ship a single transaction to a signer and we want a high degree of confidence that it will execute on-chain just like it did during simulation.